auacm/angular/src/app/blog.service.ts

```diff
@@ -0,0 +1,96 @@
+import { Injectable } from '@angular/core';
+import { Http, Request, Response, URLSearchParams, Headers } from '@angular/http';
+import { Subject } from 'rxjs';
+
+import { BlogPost, BlogAuthor } from './models/blog';
+
+@Injectable()
+export class BlogService {
+
+  constructor(private _http: Http) { }
+
+  createBlog(title: string, subtitle: string, body: string) : Promise<BlogPost> {
+    return new Promise((resolve, reject) => {
+      var params = new URLSearchParams();
+      params.append('title', title);
+      params.append('subtitle', subtitle);
+      params.append('body', body);
+
+      var headers = new Headers();
+      headers.append('Content-Type', 'application/x-www-form-urlencoded');
+
+      this._http.post(`/api/blog`, params.toString(), { headers: headers }).subscribe((res: Response) => {
+        if (res.status === 200) {
+          resolve(res.json().data);
+        } else {
+          resolve(undefined);
+        }
+      }, (err: Response) => {
+        resolve(undefined);
+      })
+    });
+  };
+
+  updateBlogPost(postId: number, title: string, subtitle: string, body: string) : Promise<BlogPost> {
+    return new Promise((resolve, reject) => {
+      var params = new URLSearchParams();
+      params.append('title', title);
+      params.append('subtitle', subtitle);
+      params.append('body', body);
+
+      var headers = new Headers();
```

Comment:
I see you reuse the same headers variable, with nothing but the content type.
It may be better to have a defaultHeader variable that you can reference in your services so you can change this much more easily if necessary.